File: android/app/src/main/java/com/bap/unsplash/modules/WallpaperModule.java
package com.bap.unsplash.modules;

import android.content.Intent;
import androidx.annotation.NonNull;
import com.bap.unsplash.services.WallpaperService;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;

public class WallpaperModule extends ReactContextBaseJavaModule {
    static String MODULE_NAME = "WallpaperModule";
    ReactApplicationContext context;

    public WallpaperModule(ReactApplicationContext context) {
        super(context);
        this.context = context;
    }

    @NonNull
    @Override
    public String getName() {
        return MODULE_NAME;
    }

    @ReactMethod
    public void setWallpaper(String path) {
        Intent intent = new Intent(context, WallpaperService.class);
        intent.setAction(WallpaperService.ACTION_SET_WALLPAPER);
        intent.putExtra("path", path);
        context.startService(intent);
    }

    @ReactMethod()
    public void setWallpaperFromStream(String url) {
        Intent intent = new Intent(context, WallpaperService.class);
        intent.setAction(WallpaperService.ACTION_SET_WALLPAPER_FROM_STREAM);
        intent.putExtra("url", url);
        context.startService(intent);
    }

    @ReactMethod
    public void downloadWithService(String url){
        Intent intent = new Intent(context, WallpaperService.class);
        intent.setAction(WallpaperService.ACTION_DOWNLOAD);
        intent.putExtra("url", url);
        context.startService(intent);
    }

    @ReactMethod
    public void download(String url){

    }
}

File: android/app/src/main/java/com/bap/unsplash/modules/WallpaperModule.kt
package com.bap.unsplash.modules

import android.app.DownloadManager
import android.content.Context
import android.content.Intent
import com.bap.unsplash.services.WallpaperService
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod

class WallpaperModule(private val context: ReactApplicationContext) : ReactContextBaseJavaModule(context) {

    companion object {
        const val MODULE_NAME = "WallpaperModule"
    }
    override fun getName() = MODULE_NAME

    @ReactMethod
    fun setWallpaper(path: String) {
        val intent = Intent(context, WallpaperService::class.java)
        intent.action = WallpaperService.ACTION_SET_WALLPAPER
        intent.putExtra("path", path)
        context.startService(intent)
    }

    @ReactMethod
    fun setWallpaperFromStream(url: String) {
        val intent = Intent(context, WallpaperService::class.java)
        intent.action = WallpaperService.ACTION_SET_WALLPAPER_FROM_STREAM
        intent.putExtra("url", url)
        context.startService(intent)
    }
}